src/new_index/mempool.rs

```diff
@@ -520,6 +491,55 @@ impl Mempool {
             .get(asset_id)
             .map_or_else(|| vec![], |entries| self._history(entries, limit))
     }
+
+    pub fn download_new_mempool_txs(
+        daemon: &Daemon,
+        old_txids: &HashSet<Txid>,
+        new_txids: &HashSet<Txid>
+    ) -> Result<Vec<Transaction>>{
+        let txids: Vec<&Txid> = (*new_txids).difference(old_txids).collect();
+        let transactions = daemon.gettransactions(&txids)
```

Comment:
`daemon.gettransactions(..).chain_err(..)` could be used directly as the return value